File: src/main/javabak/ProcessedMod.java
package redcrafter07.processed;

import net.minecraft.resources.ResourceLocation;
import net.neoforged.bus.api.IEventBus;
import net.neoforged.fml.ModContainer;
import net.neoforged.fml.common.Mod;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import redcrafter07.processed.block.ModBlocks;
import redcrafter07.processed.block.tile_entities.ModTileEntities;
import redcrafter07.processed.gui.ModMenuTypes;
import redcrafter07.processed.item.ModDataComponents;
import redcrafter07.processed.item.ModItemGroup;
import redcrafter07.processed.item.ModItems;

/**
 * Main mod class. Should be an `object` declaration annotated with `@Mod`.
 * The modid should be declared in this object and should match the modId entry
 * in neoforge.mods.toml.
 * <p>
 * An example for blocks is in the `blocks` package of this mod.
 */

@Mod(ProcessedMod.ID)
public class ProcessedMod {
    public static final String ID = "processed";

    // the logger for our mod
    public static final Logger LOGGER = LogManager.getLogger(ID);

    public static ResourceLocation rl(String path) {
        return ResourceLocation.fromNamespaceAndPath(ID, path);
    }

    public ProcessedMod(IEventBus modEventBus, ModContainer modContainer) {
        ModBlocks.BLOCKS.register(modEventBus);
        ModItems.ITEMS.register(modEventBus);
        ModItemGroup.CREATIVE_MODE_TABS.register(modEventBus);
        ModTileEntities.BLOCK_TYPES.register(modEventBus);
        ModDataComponents.DATA_COMPONENTS.register(modEventBus);
        ModMenuTypes.MENUS.register(modEventBus);
        Attachments.ATTACHMENT_TYPES.register(modEventBus);
    }
}

File: src/main/kotlin/redcrafter07/processed/ProcessedMod.kt
package redcrafter07.processed

import net.minecraft.resources.ResourceLocation
import net.neoforged.bus.api.IEventBus
import net.neoforged.fml.ModContainer
import net.neoforged.fml.common.Mod
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import redcrafter07.processed.block.ModBlocks
import redcrafter07.processed.block.tile_entities.ModTileEntities
import redcrafter07.processed.covers.covers.ModCovers
import redcrafter07.processed.entity.ModEntities
import redcrafter07.processed.fluid.ModFluids
import redcrafter07.processed.gui.ModMenuTypes
import redcrafter07.processed.items.ModDataComponents
import redcrafter07.processed.items.ModItemGroup
import redcrafter07.processed.items.ModItems
import redcrafter07.processed.network.RPCFunctions
import redcrafter07.processed.particles.ModParticles
import redcrafter07.processed.recipe.ModRecipes

fun rl(path: String): ResourceLocation = ResourceLocation.fromNamespaceAndPath(ProcessedMod.ID, path)

@Mod(ProcessedMod.ID)
class ProcessedMod(bus: IEventBus, modContainer: ModContainer) {
    companion object {
        const val ID = "processed"
        val LOG: Logger = LoggerFactory.getLogger(ID)
    }

    init {
        assert(ID == modContainer.modId)

        ModEntities.ENTITIES.register(bus)
        ModParticles.PARTICLES.register(bus)
        ModBlocks.BLOCKS.register(bus)
        ModCovers.COVERS.register(bus)
        ModItems.ITEMS.register(bus)
        ModItemGroup.CREATIVE_MODE_TABS.register(bus)
        ModTileEntities.BLOCK_TYPES.register(bus)
        ModDataComponents.DATA_COMPONENTS.register(bus)
        ModMenuTypes.MENUS.register(bus)
        Attachments.ATTACHMENT_TYPES.register(bus)
        ModRecipes.RECIPE_SERIALIZERS.register(bus)
        ModRecipes.RECIPE_TYPES.register(bus)
        ModFluids.FLUID_TYPES.register(bus)
        ModFluids.FLUIDS.register(bus)
        RPCFunctions.register(bus)

        LOG.info("Loaded processed :3")
    }
}